src/EnergyPlus/SingleDuct.cc

```diff
@@ -2358,9 +2358,9 @@ namespace SingleDuct {
 						if ( PltSizHeatNum > 0 ) {
 							CoilInTemp = TermUnitFinalZoneSizing( CurZoneEqNum ).DesHeatCoilInTempTU;
 							DesMassFlow = StdRhoAir * TermUnitSizing( CurZoneEqNum ).AirVolFlow;
-							DesZoneHeatLoad = CalcFinalZoneSizing( CurZoneEqNum ).DesHeatLoad * CalcFinalZoneSizing( CurZoneEqNum ).HeatSizingFactor;
```

Comment:
But CalcFinalZoneSizing is the intermediate sizing array that is copied to FinalZoneSizing at the end of the sizing calcs. No model should be using the Calc* arrays.

    } else if ( SELECT_CASE_var == EndZoneSizingCalc ) {
        line 2593:
        for ( std::size_t i = 0; i < FinalZoneSizing.size(); ++i ) {
            auto & z( FinalZoneSizing[ i ] );
            auto & c( CalcFinalZoneSizing[ i ] );
            z.CoolDesDay = c.CoolDesDay;
            z.HeatDesDay = c.HeatDesDay;

Then later at line 2900 is where the zone temp got updated and stored in the FinalZoneSizing array, that's why the reheat coil never saw the non-zero temperature.

    // initialize sizing conditions if they have not been set (i.e., no corresponding load) to zone condition
    if ( FinalZoneSizing( CtrlZoneNum ).ZoneTempAtHeatPeak == 0.0 ) {

If you really want this backed out then I can do that. However, searching for ZoneTempAtHeatPeak I find it in 5 zone models. It's mostly used correctly from the FinalZoneSizing array. What really needs to happen is that the use of Calc* in the models needs to be changed to FinalZoneSizing. It would be smarter to do that now and get that change in with this one.